src/integ_test/java/games/strategy/engine/lobby/server/db/BannedMacControllerIntegrationTest.java

```diff
@@ -72,9 +82,58 @@ public void testBanMacUpdate() {
     assertEquals(banUntil, macDetails2.getSecond().toInstant());
   }
 
+  @Test
+  public void testBanMacUpdatesModerator() {
+    banMacForSeconds(Long.MAX_VALUE, moderator);
```

Comment:
Has it been considered to decouple entirely the concepts of ban expiration and permanently banned?

Reply:
Yes, I remember discussing with @RoiEXLab using something other than a `NULL` ban expiration to represent a permanent ban (e.g. just using a ban duration in the "far future" to represent a permanent ban).  However, I don't think we ever agreed on how to do it.

Is your concern simply making the test code more readable here?

Reply:
A couple of concerns, I think primary concerns are SRP and avoiding magic constants. 

For considerations, the suggestion would be to have two distinct values to represent the two concepts. EG:  `<banned, banned_until>` instead of just  `<banned_until> `

So when looking for banned users, the query would be something like:
```
select * from user where banned == false || banned_until > now();
```

Reply:
@DanVanAtta We originally discussed whether it makes sense to treat the maximum values as "forever" because currently NULL in the database means forever, especially because postgres has a timestamp value representing the concept of infinity